tests/api/printf.cpp

```diff
@@ -32,6 +32,20 @@ TEST_F(WithCommandQueueAndPrintf, SimplePrintf) {
     ASSERT_STREQ(m_printf_output.c_str(), message);
 }
 
+TEST_F(WithCommandQueueAndPrintf, SimpleFormatedPrintf) {
+    const char message[] = "";
+    char source[512];
+    sprintf(source, "kernel void test_printf() { printf(\"%%s\", \"\"); }");
```

Comment:
I don't get the point of using `sprintf` if they are no arguments after the format.